eclipse-collections/src/main/java/org/eclipse/collections/impl/bimap/mutable/AbstractMutableBiMap.java

```diff
@@ -398,7 +401,7 @@ public V updateValue(K key, Function0<? extends V> factory, Function<? super V,
     }
 
     @Override
-    public Set<K> keySet()
+    public MutableSet<K> keySet()
     {
         return new KeySet();
```

Comment:
@johnjstuart do you know why this class does not simply return `delegate.keySet()` here?